### PR TITLE
docs: update architecture.md

### PR DESCRIPTION
**Pull request by jenworthington**
First editing pass of Architecture docs

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -1,62 +1,56 @@
 # SC4S Architectural Considerations
 
-Certain key architectural considerations provide performant and reliable syslog
-data collection.  These
-recommendations are not specific to Splunk Connect for Syslog, but rather stem from the syslog protocol and age.
+SC4S provides performant and reliable syslog data collection.  When you are planning your configuration, review the following architectural considerations. These recommendations pertain to the Syslog protocal and age, and are not specific to Splunk Connect for Syslog.
```
Comment (mstopa-splunk) [inline docs/architecture.md:3]:
```suggestion
SC4S provides performant and reliable syslog data collection.  When you are planning your configuration, review the following architectural considerations. These recommendations pertain to the Syslog protocol and age, and are not specific to Splunk Connect for Syslog.
```

File: docs/architecture.md [hunk 2]
```diff
@@ -1,62 +1,56 @@
...
+The syslogd daemon optimally uses UDP for log forwarding to reduce overhead. This is because UDP's streaming method does not require the overhead of establishing a network session. 
 UDP also reduces network load on the network stream with no required receipt verification or window adjustment.
-Although TCP uses ACKS and there should not be data loss, loss cann occur when:
+
+TCP uses acknowledgements (ACKS) to avoid data loss, however, loss can still occur when:
 
 * The TCP session is closed: Events published while the system is creating a new session are lost. (Closed Window Case)
-* The remote side is busy and can not ACK fast enough: Events are lost due to a full local buffer.
-* A single ACK is lost by the network and the client closes the connection: Local and remote buffer are lost.
+* The remote side is busy and cannot acknowledge fast enough: Events are lost due to a full local buffer.
```
Comment (rjha-splunk) [inline docs/architecture.md:44]: Can we specify ACK somehow as acknowledgement signal

File: docs/architecture.md [hunk 3]
```diff
@@ -1,62 +1,56 @@
...
+* A single acknowledgement is lost by the network and the client closes the connection: Local and remote buffer are lost.
```
Comment (rjha-splunk) [inline docs/architecture.md:45]: Can we specify ACK somehow as acknowledgement signal

**Review by rjha-splunk — COMMENTED** [review]
Kindly check the comments @jenworthington

**Comment by mstopa-splunk** [comment]
@jenworthington all good thank you, please see the comments above

File: docs/architecture.md (hunk 2 above)
Comment (jenworthington) [reply docs/architecture.md]: i replaced "acknowledgement" with "acknowledgement signal". I can also change it back to ACKS if you feel that works better for the audience.

**Comment by mstopa-splunk** [comment]
solved by https://github.com/splunk/splunk-connect-for-syslog/pull/2412

**Pull request closed without merging**